test/export/test_export.py

```diff
@@ -526,6 +526,24 @@ def forward(self, x):
         ):
             torch.export.export(M(), (torch.randn(2, 3),), strict=False)
 
+    def test_state_primitives(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.x = 1
+                self.y = {"x": 2}
+
+            def forward(self, x):
+                self.x = self.x + 3
+                self.y["x"] = self.y["x"] + 4
+                return x + self.x + self.y["x"]
+
+        ep = torch.export.export(M(), (torch.randn(2, 3),), strict=False)
```

Comment:
Do you want to add an assertion that after this `M().x` and `M().y` stays the same?

Reply:
They doesn't stay the same. I'm not restoring anything after export, only after the metadata collection pass. Whether to restore the state of the original module or not is an orthogonal design decision. 

(This doesn't affect the exported program, of course, because the observed state is burned in.)

Reply:
ooh ok